Avoid duplicated items

erodac opened the pull request:
Use of Set collection on the vertex routing paths instead of ArrayList to avoid any duplicated items on the collection. To modify this and ensure the exclusivity of each item, was necessary to override the equals() and hashCode() methods of RoutingOption class. An unitary test was successfully done.

erodac committed:
Use of Set collection on RBR.routingPathForVertex.
Description:The attribute was modified from ArrayList type to Set collection to avoid any possible duplicated items insertion.
Commit:Mod
TaskNumber:#32

erodac committed:
Unitary test for the use of Set collection.
Description:The override of equals() and hashcode() methods of RoutingOption was successfully tested.
Commit:Add
TaskNumber:#32

jmferreiratech commented on src/rbr/RBR.java:
```diff
@@ -343,8 +343,7 @@ private boolean mergeUnitary(Vertex router) {
 
 	private void addRoutingPath(Vertex v, Set<Character> ip, Vertex dst, Set<Character> op) {
```
Could you please add a Ref commit to get ride of this method and replace the calls made to it by "routingPathForVertex.get(v).add(new RoutingOption(ip, dst, op));"?

jmferreiratech commented on src/rbr/RoutingOption.java:
```diff
@@ -41,4 +41,13 @@ public boolean equals(Object _that) {
+		int result = ip.hashCode();
+		result = 37 * result + op.hashCode();
+		result = 37 * result + dst.hashCode();
+		
```
Remove that blank line.

jmferreiratech commented on tests/rbr/RoutingPathsDuplicationTest.java:
```diff
@@ -0,0 +1,31 @@
+package rbr;
+
+import java.util.HashSet;
```
Squash these imports with *

jmferreiratech commented on tests/rbr/RoutingPathsDuplicationTest.java:
```diff
@@ -0,0 +1,31 @@
+	public void equalityTest(){
+		Set<Character> ip = new HashSet<>();
+		Set<Character> op = new HashSet<>();
+		RoutingOption option1, option2;
```
Merge declaration and initialization.

jmferreiratech commented on tests/rbr/RoutingPathsDuplicationTest.java:
```diff
@@ -0,0 +1,31 @@
+
+	@Test
+	public void equalityTest(){
+		Set<Character> ip = new HashSet<>();
```
Declare variables more closer to their use.

jmferreiratech commented on tests/rbr/RoutingPathsDuplicationTest.java:
```diff
@@ -0,0 +1,31 @@
+		option2 = new RoutingOption(ip, sw, op);
+		
+		Assert.assertEquals("option2: " + option2.hashCode() + " option1: " + option1.hashCode(), option2, option1);
+		Assert.assertEquals(option1.hashCode(), option2.hashCode());
```
This is another test.

jmferreiratech commented on tests/rbr/RoutingPathsDuplicationTest.java:
```diff
@@ -0,0 +1,31 @@
+		option1 = new RoutingOption(ip, sw, op);
+		option2 = new RoutingOption(ip, sw, op);
+		
+		Assert.assertEquals("option2: " + option2.hashCode() + " option1: " + option1.hashCode(), option2, option1);
```
The message has no relation with the assert itself.

jmferreiratech commented on tests/rbr/RoutingPathsDuplicationTest.java:
```diff
@@ -0,0 +1,31 @@
+		Assert.assertEquals("option2: " + option2.hashCode() + " option1: " + option1.hashCode(), option2, option1);
+		Assert.assertEquals(option1.hashCode(), option2.hashCode());
+	}
+
```
Remove this blank line.

jmferreiratech commented on tests/rbr/RoutingPathsDuplicationTest.java:
```diff
@@ -0,0 +1,31 @@
+import rbr.RoutingOption;
+import util.Vertex;
+
+public class RoutingPathsDuplicationTest {
```
Name it only RoutingPathTest

erodac committed:
Removed unnecessary method

Description:The method addRoutingPath() doesn't need to exist.

Commit:Ref

TaskNumber:#32

erodac commented:
All changes were done.
